src/main/java/tennisclub/dao/EventDaoImpl.java

```diff
@@ -0,0 +1,79 @@
+package tennisclub.dao;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import tennisclub.entity.Court;
+import tennisclub.entity.Event;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.time.LocalDateTime;
+import java.util.List;
+
+/**
+ * @author Miroslav Demek
+ */
+@Repository
+@Transactional
```

Comment:
This makes every method transactional, which we don't want (calling multiple methods would cause multiple transactions). The transaction should be managed by an upper level to allow multiple DAO operations within a single transaction.